src/rrdp-iostat/rrdp_iostat.ml

```diff
@@ -28,12 +28,36 @@ let with_xc f = Xenctrl.with_intf f
 
 (* Return a list of (domid, uuid) pairs for domUs running on this host *)
 let get_running_domUs xc =
-  let open Xenctrl in
-  Xenctrl.domain_getinfolist xc 0 |> List.map (fun di ->
-      let domid = di.domid in
-      let uuid = Uuid.to_string (Uuid.uuid_of_int_array di.handle) in
-      (domid, uuid)
-    ) |> List.filter (fun x -> fst x <> 0)
+  let metadata_of_domain di =
+    let open Xenctrl in
+    let domid = di.domid in
+    let uuid = Uuid.(to_string (uuid_of_int_array di.handle)) in
+
+    (* Actively hide migrating VM uuids, these are temporary and xenops
+       writes the original and the final uuid to xenstore *)
+    let uuid_from_key key =
+      let path = Printf.sprintf "/vm/%s/%s" uuid key in
+      try
+        Xenstore.(with_xs (fun xs -> xs.read path))
```

Comment:
This will connect and disconnect from xenstore once for each VM. We should connect only once outside of the loop.